src/main/scala/is/hail/annotations/StagedRegionValueBuilder.scala

```diff
@@ -47,9 +47,9 @@ class StagedRegionValueBuilder private(val mb: MethodBuilder, val typ: Type, var
 
   def currentOffset: Code[Long] = {
     typ match {
-      case _: TBaseStruct => elementsOffset
-      case _: TArray => elementsOffset
-      case _ => startOffset
+      case _: TBaseStruct => elementsOffset.load()
```

Comment:
You can get around this by adding another def to `asm4s/package.scala` that mimics the one for `LocalRef`:

```scala
  implicit def toCode[T](f: LocalRef[T]): Code[T] = f.load()
```